internal/storage/sql/db.go

```diff
@@ -34,6 +34,13 @@ func Open(cfg config.Config, opts ...Option) (*sql.DB, Driver, error) {
 	if cfg.Database.MaxOpenConn > 0 {
 		sql.SetMaxOpenConns(cfg.Database.MaxOpenConn)
 	}
+
+	// if we're using sqlite, we need to set always set the max open connections to 1
+	// see: https://github.com/mattn/go-sqlite3/issues/274
+	if driver == SQLite {
+		sql.SetMaxOpenConns(1)
+	}
```

Comment:
How about we also log a warning suggesting we're going to ignore any attempts to manually configure this because of the driver choice:

```suggestion
	var maxOpenConn int 
	if cfg.Database.MaxOpenConn > 0 {
		maxOpenConn = cfg.Database.MaxOpenConn
	}

	// if we're using sqlite, we need to set always set the max open connections to 1
	// see: https://github.com/mattn/go-sqlite3/issues/274
	if driver == SQLite {
		if maxOpenConn > 1 {
			log.Warning("Ignoring config.db.max_open_conn due to driver limitation (sqlite)", zap.String("attempted_max_conn", maxOpenConn))
		}
		
		maxOpenConn = 1
	}
	
	sql.SetMaxOpenConns(maxOpenConn)
```